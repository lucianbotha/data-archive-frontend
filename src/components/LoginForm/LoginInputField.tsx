import * as React from "react";
import InputField from "../basicComponents/InputField";

/*
The login input field responsible for user input.

Properties:
-----------
label
  The label which detail the user input field
name
  The name that uniquecly identifies the input field
onChange
  The fuction that is responsible for updating the input field value when changed
placeholder
  It gives an example of the required value
type
  The type of the input field e.g. text, password, etc.
value
  The value of the input field
*/

interface ILoginInputProps {
  error?: string;
  label?: string;
  loading?: boolean;
  name?: string;
  onChange?: any;
  placeholder?: string;
  type?: string;
  value?: string;
}

const LoginInputField = (props: ILoginInputProps) => {
  const { error, label, loading, name, onChange, type, value } = props;
  return (
    <div className={"field"}>
      <label htmlFor={`${name}+${label}`} className={"label"}>
        {label}:
      </label>
      <div className={"control is-child"}>
        <InputField
          data-test="target-name-input"
          disabled={loading}
          name={name}
          value={value || ""}
          error={error}
          onChange={onChange}
          type={type}
        />
      </div>
    </div>
  );
};

export default LoginInputField;
