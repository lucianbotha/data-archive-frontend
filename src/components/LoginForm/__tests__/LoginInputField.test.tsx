import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import LoginInputField from "../LoginInputField";

describe("LoginInputField Component", () => {
  it("renders the LoginInputField having unpopulated props with no errors", () => {
    const wrapper = shallow(<LoginInputField />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the LoginInputField having populated props with no errors", () => {
    const wrapper = shallow(
      <LoginInputField
        name={"username"}
        label={"Username"}
        value={""}
        onChange={() => ({})}
        placeholder={""}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
