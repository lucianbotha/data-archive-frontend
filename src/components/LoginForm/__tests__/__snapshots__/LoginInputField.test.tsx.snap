// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginInputField Component renders the LoginInputField having populated props with no errors 1`] = `
<div
  className="field"
>
  <label
    className="label"
    htmlFor="username+Username"
  >
    Username
    :
  </label>
  <div
    className="control is-child"
  >
    <input
      className="username input"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder=""
      value=""
    />
  </div>
</div>
`;

exports[`LoginInputField Component renders the LoginInputField having unpopulated props with no errors 1`] = `
<div
  className="field"
>
  <label
    className="label"
    htmlFor="undefined+undefined"
  >
    :
  </label>
  <div
    className="control is-child"
  >
    <input
      className="undefined input"
    />
  </div>
</div>
`;
